extensions/typescript/src/features/taskProvider.ts

```diff
@@ -0,0 +1,110 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
I like the idea of inferring the tsconfig.json to use from the open active TS file. However if no file is open I think we should still provide a reasonable command and not just look into the workspace root. We should search for all tsconfig.json and return all of them. If a build is trigger the task runner will run all of them

Regarding activation: I have a similar problem in ESLint where I need to active the plugin to simply being able to listen to config changes. This is all I do in activate. In addition I listen to open event and if a JS files opens I start the ESLint server by hand. So basically I have an activate and a realActivate. We could do the same for TS and start the tsserver when we get a open event for a TS file in the listener registered in activate.